napari/layers/labels/labels.py

```diff
@@ -732,15 +710,46 @@ def mode(self):
         Mode.ERASE: 'circle',
     }
 
+    @staticmethod
+    def _mode_help_string(mode: Mode) -> str:
+        """
+        Return string informing about shortcuts
+        that will enable alternative layer mode
+        """
+        from . import _labels_key_bindings as kb
+
+        help_li = []
+        shortcuts = get_settings().shortcuts.shortcuts
+
+        for func, mode_ in (
+            (kb.activate_label_erase_mode, Mode.ERASE),
+            (kb.activate_paint_mode, Mode.PAINT),
+            (kb.activate_fill_mode, Mode.FILL),
+            (kb.activate_label_picker_mode, Mode.PICK),
+            (kb.activate_label_pan_zoom_mode, Mode.PAN_ZOOM),
+        ):
+            if mode == mode_:
+                continue
+            action_name = f"napari:{func.__name__}"
+            desc = action_manager._actions[action_name].description.lower()
+            help_li.append(
+                trans._(
+                    "use <{shortcut}> for {desc}",
+                    shortcut=shortcuts[action_name][0],
+                    desc=desc,
+                )
+            )
+        return ", ".join(help_li)
+
     @mode.setter
     def mode(self, mode: Union[str, Mode]):
         mode, changed = self._mode_setter_helper(mode, Mode)
         if not changed:
             return
 
-        self.help = _FWD_SHAPE_HELP[mode]
+        self.help = self._mode_help_string(mode)
 
-        if mode in (Mode.PAINT, Mode.ERASE):
+        if mode in {Mode.PAINT, Mode.ERASE}:
```

Comment:
Is there a reason for this change? Just curious, cause I don't see what it would change.

Reply:
change from a tuple to set. It is a little more pythonic.

Reply:
Why? Is it faster? Or just more "correct"? I feel like I've only ever seen the tuple version :)

Reply:
In theory it's faster, O(1), to check whether something is in a set vs a tuple (O(N)). For a 2-tuple? Probably about the same. I would agree with @Czaki that the "preferred" form for `if x in xs` is for xs to be a set, unless it is *already* a list or a tuple for some prior reason.

Reply:
Here you go:

```
In [184]: %timeit 1 in {1, 2}
83.2 ns ± 3.74 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [185]: %timeit 1 in (1, 2)
70.4 ns ± 3.14 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [186]: %timeit 1 in (2, 1)
104 ns ± 8.75 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [187]: %timeit 1 in {2, 1}
83.6 ns ± 2.71 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

Reply:
So the advantage is measureable even for two elements. The only time tuple is faster is if you hit the very first element. tbh I'm surprised by this and it points to some serious optimisation of sets/dicts in Python!

Reply:
Just more correct. With a set of sizes, two the speeds should be the same. For the larger one, it may have an impact. 

I change it because of the linter highlight.

Reply:
also
```python
In [1]: _set = set(range(1000))

In [2]: _tup = tuple(range(1000))

In [3]: %timeit 100 in _set
27.9 ns ± 0.0384 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [4]: %timeit 100 in _tup
632 ns ± 2.39 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

Reply:
Wow, I never considered that!

Worth noting that constructing a set seems to be slower:
```py
In [1]: %timeit set(range(1000))
21.2 µs ± 2.07 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

In [2]: %timeit tuple(range(1000))
12.9 µs ± 155 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

So unless the set exists already, this time should also be accounted for (which @jni already did for the 2-tuple case).

```py
In [10]: %timeit 10 in set(range(10))
362 ns ± 8.62 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [11]: %timeit 10 in tuple(range(10))
325 ns ± 9.09 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

So in practice, a tuple might be better for this kind of case :P

Reply:
Yes. But

```python
In [1]: %timeit 10 in {0,1,2,3,4,5,6,7,8,9}
22.4 ns ± 0.918 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [2]: %timeit 10 in (0,1,2,3,4,5,6,7,8,9)
111 ns ± 0.54 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

As build set from the constructor and using literal constructor is different.

Reply:
Damn, crazy!

Reply:
Some longer comparisons:
```python
In [1]: %timeit 10 in {0,1,2,3,4,5,6,7,8,9}
22.1 ns ± 0.0787 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [2]: %timeit 10 in (0,1,2,3,4,5,6,7,8,9)
114 ns ± 3.87 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [3]: %timeit 10 in {i for i in range(10)}
537 ns ± 9.94 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [4]: %timeit 10 in [i for i in range(10)]
543 ns ± 7.96 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [5]: %timeit 10 in set(range(10))
334 ns ± 1.06 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [6]: %timeit 10 in list(range(10))
357 ns ± 5.39 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [7]: %timeit 10 in tuple(range(10))
320 ns ± 3.06 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

where `set(range(10)) is faster than `{i for i in range(10)}`